.github/workflows/create-project.yml

```diff
@@ -0,0 +1,36 @@
+name: "Create Project"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - '[0-9]+.[0-9]+.x'
+      - 'refs/pull/*'
+    tags:
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        php_version:
+          - '@latest'
+          - '@lowest'
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    steps:
+      - name: Checkout CI environment
+        uses: laminas/laminas-continuous-integration-action@v1
+        with:
+          php: ${{ matrix.php_version }}
+      - name: Checkout sourcecode
+        uses: actions/checkout@v3
+      - name: Test create-project
+        run: |
+          cd ..
+          yes 1 | composer create-project mezzio/mezzio-skeleton test-new-project \
+              --repository='{"type": "path", "url": "./mezzio-skeleton"}' --stability=dev
```

Comment:
I left a way to reproduce the behavior using a `git archive` in https://github.com/mezzio/mezzio-skeleton/pull/114#issuecomment-1373857762, but I found that this way I could test it using a real `composer create-project` command.

This time the project is created based on the local path instead of the version published on Packagist, which can be verified by truncating the file path in `OptionalPackages::preparePhpunitConfig()`.

<img width="803" alt="スクリーンショット 2023-01-07 3 12 21" src="https://user-images.githubusercontent.com/822086/211073057-518880e2-cd9a-40a1-a7bd-379607952758.png">

For some reason composer creates `test-new-project` as a symlink instead of creating a new directory, which is fine for a working check.